lightning/src/ln/channel.rs

```diff
@@ -293,6 +291,7 @@ pub(super) struct Channel<ChanSigner: ChannelKeys> {
 	pending_inbound_htlcs: Vec<InboundHTLCOutput>,
 	pending_outbound_htlcs: Vec<OutboundHTLCOutput>,
 	holding_cell_htlc_updates: Vec<HTLCUpdateAwaitingACK>,
+	payment_preimages: HashMap<PaymentHash, PaymentPreimage>,
```

Comment:
I don't see where this is pruned from? In ChannelMonitor we remove entries when we're sure we no longer need them (ie all non-revoked commitment txn don't have any HTLC with the given hash), and otherwise this grows unbounded with every HTLC we relay.

Reply:
Oh, I thought that in lightning you want to hold on to (at least some) preimages in the case where we're paying an invoice, as "proof of payment."

Reply:
Right, as a sender you may (though we push that logic onto our users), but an intermediate node shouldn't care.